src/ark.ts

```diff
@@ -1,107 +1,247 @@
 import { Transport as LedgerTransport } from "@ledgerhq/hw-transport";
 import { Transport } from "./contracts";
-import { splitPath, splitToChunks, writePathsToBuffer } from "./utils";
-
+import {
+	parseBip32Path,
+	splitApduChunks,
+	writeBip32ElementsToBuffer,
+} from "./utils";
+	
+/**
+ * APDU Header Flags
+ *
+ * Describes the APDU Class, Instruction-Type, Parameter 1, and Parameter 2.
+ *
+ * APDU Header:  ({ CLA + INS + P1 + P2 })
+ * - CLA:  Apdu Class
+ * - INS:  Instruction Type
+ * - P1:   Instruction Parameter 1
+ * - P2:   Instruction Parameter 2
+ *
+ * Instruction Types:
+ * - INS_GET_PUBLIC_KEY:    Get a PublicKey from a Ledger Device
+ * - INS_GET_VERSION:       Get the ARK Application Version from a Ledger Device
+ * - INS_SIGN_TRANSACTION:  Sign a Transaction using a Ledger Device
+ * - INS_SIGN_MESSAGE:      Sign a Message using a Ledger Device
+ *
+ * App / PublicKey Context:
+ * P1: User Approval
+ * - P1_NON_CONFIRM:  Do NOT request user approval
+ * - P1_CONFIRM:      Request user approval
+ *
+ * P2: ChainCode
+ * - P2_NO_CHAINCODE:  Don't use a ChainCode
+ * - P2_CHAINCODE:     Use a Chaincode
+ *
+ * Signing Context:
+ * P1: Payload Segment
+ * - P1_SINGLE:  N(1) where N === 1
+ * - P1_FIRST:   N(1) where N > 1
+ * - P1_MORE:    N(2)..N-1 where N > 2
+ * - P1_LAST:    Nth where N > 1
+ *
+ * P2:
+ * - P2_ECDSA: Use Ecdsa Signatures
+ *
+ */
+enum ApduFlags {
+	/** APDU Class */
+	CLA = 0xe0,
+
+	/** App / PublicKey Context */
+	INS_GET_PUBLIC_KEY = 0x02,
+	INS_GET_VERSION = 0x06,
+
+	P1_NON_CONFIRM = 0x00,
+	P1_CONFIRM = 0x01,
+
+	P2_NO_CHAINCODE = 0x00,
+	P2_CHAINCODE = 0x01,
+
+	/** Signing Context */
+	INS_SIGN_TRANSACTION = 0x04,
+	INS_SIGN_MESSAGE = 0x08,
+
+	P1_SINGLE = 0x80,
+	P1_FIRST = 0x00,
+	P1_MORE = 0x01,
+	P1_LAST = 0x81,
+
+	P2_ECDSA = 0x40,
+}
+
+/**
+ * ARK Ledger Transport Class.
+ *
+ * Send APDU Instruction Payloads to a Ledger Device.
+ *
+ * - INS_GET_PUBLIC_KEY
+ * - INS_GET_VERSION
+ * - INS_SIGN_TRANSACTION
+ * - INS_SIGN_MESSAGE
+ */
 export default class ARK implements Transport {
-	readonly IDENTIFIER = 0xe0;
-	readonly OP_GET_PUBLIC_KEY = 0x02;
-	readonly OP_SIGN_TRANSACTION = 0x04;
-	readonly OP_GET_VERSION = 0x06;
-	readonly OP_SIGN_MESSAGE = 0x08;
-	readonly ALG_SECP256K1 = 0x40;
-	readonly CHUNK_SIZE = 255;
-	readonly PAYLOAD_MAX = 255 * 4;
 	private transport: LedgerTransport<any>;
 
-	constructor (transport: LedgerTransport<any>) {
+	readonly CHUNK_MAX: number = 10;
+	readonly CHUNK_SIZE: number = 255;
+	readonly PAYLOAD_MAX: number = this.CHUNK_MAX * this.CHUNK_SIZE;
+
+	/**
+	 * ARK Ledger Transport Class Ctor.
```

Comment:
I don't think this is necessary

Reply:
agreed, this one slipped though. C terms don't belong here.